### PR TITLE
Added documentation in the README and also made browser app example a little bit easier to read

### PR DESCRIPTION
**Pull request by sole**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,87 @@ dns-sd.js
...
+
+TODO: test to see if this works.
+
+### In practice
+
+This library requires that your app has access to TCP and UDP sockets, so you need to specify these in the `permissions` field in your `manifest.webapp` file:
+
+```javascript
+{
+  "tcp-socket": {},
```
Comment (justindarc) [inline README.md]: Actually, dns-sd.js doesn't need `tcp-socket` itself. Its just that in most cases, you'll need it to communicate after you've discovered a peer (e.g. using a web server).

**Comment by DougReeder** [comment]
These additional docs are very helpful for someone new to this lib.

**Comment by sole** [comment]
hey @justindarc I changed the wording a little bit. Let me know how it looks! I think it's PR-Merge material now ;-)

**Comment by sole** [comment]
@DougReeder thanks, glad they're useful! =)

**Comment by justindarc** [comment]
Thanks @sole!

**Pull request merged**